### PR TITLE
Add more info to signature error message

### PR DESCRIPTION
**Pull request by mrsmkl**
### Description

Validators keep seeing this message.
Adding the address against which the signature is checked to the error message.

Changed domain verification to check using all signer addresses like the others.

### CODE REVIEW

File: packages/contractkit/src/identity/claims/verify.ts [hunk 1]
```diff
@@ -70,37 +69,45 @@ type dnsResolverFunction = (
...
   address: string,
   dnsResolver: dnsResolverFunction = resolveTxt as any
 ) => {
   const found = await new Promise((resolve) => {
-    dnsResolver(claim.domain, (error, domainRecords) => {
+    dnsResolver(claim.domain, async (error, domainRecords) => {
       if (error) {
         console.debug(`Unable to fetch domain TXT records: ${error.toString()}`)
-        resolve(false)
+        resolve(`Unable to fetch domain TXT records: ${error.toString()}`)
```
Comment (jmrossy) [inline packages/contractkit/src/identity/claims/verify.ts]: We should be using `reject` here. Resolving truthy values that are meant to be a failures can lead to misinterpreted results in consumers.

File: packages/contractkit/src/identity/claims/verify.ts [hunk 2]
```diff
@@ -70,37 +69,45 @@ type dnsResolverFunction = (
...
                 resolve(true)
               }
             }
-          })
-        })
+          }
+        }
       }
-      resolve(false)
+      resolve(`Unknown failure with address ${address}`)
```
Comment (jmrossy) [inline packages/contractkit/src/identity/claims/verify.ts]: ditto

**Review by jmrossy — COMMENTED** [review]
Mostly looks good but promise handling needs a fix

File: packages/contractkit/src/identity/claims/verify.ts (hunk 1 above)
Comment (mrsmkl) [reply packages/contractkit/src/identity/claims/verify.ts]: Agree, but a different pattern is used here for some reason (undefined means success and returned strings are error messages). Anyway cleaned it a bit up to make it easier to understand.

**Review by gastonponti — APPROVED** [review]
LGTM

**Review by nambrot — APPROVED** [review]

**Pull request merged**